### PR TITLE
Nullif

### PR DESCRIPTION
**Pull request by azurenake**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/nullif.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
+#ifndef DORIS_FUNCTION_IFNULL_H
```
Comment (HappenLee) [inline be/src/vec/functions/nullif.cpp:1]: 这里需要补充一下协议注释

File: be/src/vec/functions/nullif.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    bool use_default_implementation_for_constants() const override { return false; }
+
+    bool use_default_implementation_for_nulls() const override { return false; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return arguments[0];
+    }
+
+    NullPresence getNullPresense(const ColumnsWithTypeAndName& args) const{
```
Comment (HappenLee) [inline be/src/vec/functions/nullif.cpp:39]: 函数名用下划线命名 ，const {有个空格

**Review by HappenLee — COMMENTED** [review]
有一些小问题，辛苦改一下哈

File: be/src/vec/functions/nullif.cpp [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        const ColumnsWithTypeAndName& arguments) const {
+        
+        size_t num_full_ordinary_columns = 0;
+        ColumnsWithTypeAndName args_without_low_cardinality(arguments);
+
+        for (ColumnWithTypeAndName& arg : args_without_low_cardinality) {
+            bool is_const = arg.column && is_column_const(*arg.column);
+            if (is_const)
+                arg.column = assert_cast<const ColumnConst&>(*arg.column).remove_low_cardinality();
+            if (!is_const) ++num_full_ordinary_columns;
```
Comment (zhangstar333) [inline be/src/vec/functions/nullif.cpp:60]: 这个后面好像没有用上它？

Comment (azurenake) [reply be/src/vec/functions/nullif.cpp:60]: num_full_ordinary_columns是没有地方用到，这段是从function.cpp copy过来的

Comment (azurenake) [reply be/src/vec/functions/nullif.cpp:60]: > 这个后面好像没有用上它？

num_full_ordinary_columns是没有地方用到，这段是从function.cpp copy过来的

Comment (HappenLee) [reply be/src/vec/functions/nullif.cpp:74]: 没有用的就先删除了吧

File: be/src/vec/functions/nullif.cpp [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+#include "vec/data_types/data_type_nothing.h"
+#include "vec/data_types/data_type_nullable.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/data_types/get_least_supertype.h"
+#include "vec/data_types/number_traits.h"
+
+namespace doris::vectorized {
+class FunctionNullIf : public IFunction {
+public:
+
```
Comment (HappenLee) [inline be/src/vec/functions/nullif.cpp:32]: 这个空格没必要哈

**Pull request merged**